kubernetes/audit.go

```diff
@@ -0,0 +1,114 @@
+package kubernetesfalc
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/api/auditregistration/v1alpha1"
+
+	"github.com/kris-nova/logger"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	APIServerNamespace = "kube-system"
+)
+
+type AuditInstaller struct {
+	k8s            *kubernetesConfigClient
+	FalcoNamespace string
+}
+
+func (i *AuditInstaller) Install(k8s *kubernetesConfigClient) error {
+	i.k8s = k8s
+	logger.Info("Configuring Kubernetes Audit Logging")
+	logger.Info("Searching for API server...")
+	pods, err := k8s.client.CoreV1().Pods(APIServerNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to list pods in kube-system Namespace: %v", err)
+	}
+	var apiServer v1.Pod
+	found := false
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, "kube-apiserver") {
+			found = true
+			logger.Success("Found API server...")
+			apiServer = pod
+			break
+		}
+	}
+	if !found {
+		return fmt.Errorf("unable to find kube-apiserver")
+	}
+
+	// -----------------------------------------------------------------------------------------------------------------
+	//
+	// Enable Audit Flags
+	logger.Info("Setting new API server flags...")
+	apiServer.Spec.Containers[0].Command = append(apiServer.Spec.Containers[0].Command,
+		"--tls-private-key-file=/etc/kubernetes/pki/apiserver.key")
+	apiServer.Spec.Containers[0].Command = append(apiServer.Spec.Containers[0].Command,
+		"--feature-gates=DynamicAuditing=true")
+	apiServer.Spec.Containers[0].Command = append(apiServer.Spec.Containers[0].Command,
+		"--runtime-config=auditregistration.k8s.io/v1alpha1=true")
+
+	//_, err = i.k8s.client.CoreV1().Pods(APIServerNamespace).Update(&apiServer)
+	//if err != nil {
+	//	return fmt.Errorf("unable to update API server: %v", err)
+	//}
+
+	// TODO - Generate YAML and patch the server manifest
```

Comment:
There is an `ssh` package we use in `kubicorn` we could use to patch the API server manifest. 

We can at least generate the YAML here and wipe it, so we don't have to try to modify the existing one. We can just replace. 

AuditSink won't work until we fix this.